In [42]:
import numpy as np
import pandas as pd
import pyAgrum as gum
import pyAgrum.lib.notebook as gnb

## We are given this database coming from the historical records of a hospital

In [43]:
df = pd.read_csv("sample")
df

smoker  treatment  dead
0            0          0     0
1            1          1     0
2            1          1     0
3            0          1     0
4            0          0     0
...        ...        ...   ...
999995       0          0     0
999996       1          1     0
999997       0          0     0
999998       0          0     0
999999       0          0     0

[1000000 rows x 3 columns]

## And we are asked: 
## Question: How much will number of deads change if we decide treat everybody from now on with respect to not treating anyone? 

In [44]:
treated = df[df["treatment"]==1]

In [45]:
untreated = df[df["treatment"]==0]

In [46]:
percentage_dead_treated = treated["dead"].mean()
percentage_dead_treated

0.32080451312239394

In [47]:
percentage_dead_untreated= untreated["dead"].mean()
percentage_dead_untreated

0.23595664229896648

In [48]:
percentage_dead_treated - percentage_dead_untreated

0.08484787082342746

## Response: The percentage of deads will increase a 8.5% if we decide to treat everybody (it will jump from 23.56% to 32.08%). Consequently, we SHOULD NOT TREAT our population.

## Unless... we were not too careful when putting together the numbers...